includes/reader-revenue/stripe/class-stripe-connection.php

```diff
@@ -786,38 +785,112 @@ public static function handle_donation( $config ) {
 				return $response;
 			}
 
-			// Attach the Payment Method ID to the customer.
-			// A new payment method is created for each one-time or first-in-recurring
-			// transaction, because the payment methods are not stored on WP.
-			$stripe->paymentMethods->attach( // phpcs:ignore
-				$payment_method_id,
-				[ 'customer' => $customer['id'] ]
-			);
-			// Set the payment method as the default for customer's transactions.
-			$stripe->customers->update(
+			$is_recurring = 'once' != $frequency;
+
+			// In this mode, WC will create a subscription for the customer, instead of using
+			// native Stripe subscriptions.
+			$is_wc_first         = false;
+			$is_wc_first_enabled = defined( 'NEWSPACK_USE_WC_SUBSCRIPTIONS_WITH_STRIPE_PLATFORM' ) && NEWSPACK_USE_WC_SUBSCRIPTIONS_WITH_STRIPE_PLATFORM;
+			if ( $is_wc_first_enabled && Donations::is_woocommerce_suite_active() ) {
+				$gateways = WooCommerce_Configuration_Manager::get_payment_gateways( true );
+				// Check if Stripe (Credit Card) is the only payment gateway enabled.
+				if ( 1 === count( $gateways ) && isset( $gateways['stripe'] ) ) {
```

Comment:
Why limit this behavior to sites that _only_ have the Stripe payment gateway enabled? Why not also create Woo subscriptions as long as it's one of the configured payment gateways?

Reply:
Good question! This was premature on my part – in the future there should be just one on-site RR platform. In this feature, SDB would render only if `Stripe (Credit Card)` is the only gateway enabled. Otherwise the other gateways would be unavailable. Reverted this change in aa1bc0991e958e87543bbdcca82860faa9365bba.